walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,198 @@
 
 package com.google.sps;
 
+import com.google.sps.algorithms.BinarySearch;
+import com.google.sps.algorithms.MergeSort;
+import com.google.sps.comparator.SortEventsByTime;
+import com.google.sps.predicate.IsIntersection;
+import com.google.sps.TimeRange;
+import java.lang.Math;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Predicate;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  /**
+   * Turns an ordered Collection of events (by time) into an ArrayList of an 
+   *    array that contains the start and end time.
+   * Also, it will remove all of the events with the same start time but will 
+   *    keep the event with the greatest duration (or latest end time) of those 
+   *    with the same start time.
+   * Also, it will remove an event that is contained within another event.
+   * Time Complexity: O(n)
+   *
+   * @param events the Collection of events that will turn into its start and 
+   *     end time
+   * @return An ArrayList that contains an array of start and end times, and 
+   *     will not have any events with the same start time.
+   */
+  public static ArrayList<int[]> eventToTime(Event[] eventsArray) 
+      throws Exception {
+    ArrayList<int[]> eventTimes = new ArrayList<>();
+    TimeRange previousEventTR = TimeRange.fromStartDuration(0, 0);
+
+    for (Event event : eventsArray) {
+      TimeRange eventTR = event.getWhen();
```

Comment:
nit: Minor edit, but for readability I'd expand `eventTR` to `eventTimeRange` and also expand TR to TimeRange in other local variables too.